### ISSUE TITLE
Pinpoint stats API requests

### ISSUE THREAD

**Issue by anniesullie** [catapult-project/catapult#4217]
The stats API (or another API to provide bisect data) should include the following:

 * Bug id
 * Data about each difference:
   * Absolute and percent delta
   * Culprit CL (enough information to look up author, commit time, description)
 * Perf dashboard test_path
 * A reference to perf dashboard alert, if possible. (We can get a unique ID from the dashboard key)

This will help with both dashboard UI and project soundwave use cases.

@dave-2 
@simonhatch 
@perezju

**Comment by dave-2** [comment]
I was thinking about having `/api/jobs` let you customize what information you want it to return. [Gerrit API](https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html) provides something similar with the `o` parameter

**Comment by anniesullie** [comment]
That sounds great! From my perspective, just needs to be a way to get at the info I requested.

**Comment by simonhatch** [comment]
Will we also need to get the stats for a specific job, ie. should both /api/jobs and /api/job accept the o parameter?

**Comment by dave-2** [comment]
I don't think we'll need it, but we'll get it for free if we pass the `o` parameter through `AsDict(self, o)`.

**Comment by simonhatch** [comment]
Yeah not sure who'll need it, the perf triage changes might, the pinpoint stats page will definitely like /api/jobs to take o, so I'll just make it available on both.

**Comment by simonhatch** [comment]
So spoke offline w/ @dave-2 since Pinpoint doesn't have master at the moment (and thus can't return a test_path), we're going to add an arbitrary job data jsonproperty that contains metadata from the dashboard (so it can have both test path and the alert it was created from).

So one small caveat that may or may not be important, if we allow people to start bisects from Pinpoint rather than the dashboard (excluding a re-run option), they won't have this metadata.

**Comment by perezju** [comment]
I think that's fine, as long as we have the `configuration`. It's also certainly fine for bisects started by manually by people to _not_ be associated with any waterfall bot or alert.

**Comment by simonhatch** [comment]
So current status of this is that you can query for nearly everything except for the summary stats about each difference.
